spanner/value.go

```diff
@@ -1899,6 +2018,33 @@ func decodeValue(v *proto3.Value, t *sppb.Type, ptr interface{}, opts ...decodeO
 			return errBadEncoding(v, err)
 		}
 		proto.Unmarshal(y, p)
+	case *NullProto:
+		if p == nil {
+			return errNilDst(p)
+		}
+		if p.ProtoVal == nil {
+			return errNilDst(p.ProtoVal)
+		}
+		if reflect.ValueOf(p.ProtoVal).Kind() != reflect.Ptr {
+			return errNotAPointer(p.ProtoVal)
+		}
+		if code != sppb.TypeCode_PROTO && code != sppb.TypeCode_BYTES {
+			return errTypeMismatch(code, acode, ptr)
+		}
+		if isNull {
+			*p = NullProto{}
+			break
+		}
+		x, err := getStringValue(v)
```

Comment:
Line 2038 to 2045 is repeated multiple times in this file. Should we extract a method for it?